msccl/language/collectives.py

```diff
@@ -105,6 +105,10 @@ def init_buffers(self):
                 output_buffer = [None] * (self.num_ranks * self.chunk_factor)
                 for ch in range(self.chunk_factor):
                     input_buffer[ch] = Chunk(r, ch, -1, r * self.chunk_factor + ch)
+                if self.create_all_chunks:
+                    for rank in range(self.num_ranks):
+                        for ch in range(self.chunk_factor):
+                            output_buffer[rank * self.chunk_factor + ch] = Chunk(rank, ch, -1, rank * self.chunk_factor + ch)
```

Comment:
The args for Chunk seems incorrect. There are some comments:
1. original_rank should always be r I think. For example, for GPU 0, the chunks in output buffer should set original rank to 0
2. For last two args in Chunk constructor, set it to -1, -1?

Reply:
This make sense, I guess you are right. BTW the same is wrong in the inplace scenario, right? I guess that I did the same mistake there.